packages/flutter_tools/test/general.shard/dart/sdk_validation_test.dart

```diff
@@ -0,0 +1,136 @@
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+
+void main() {
+  group('sdk validation', () {
+    AnalysisServer server;
+    Directory tempDir;
+
+    setUpAll(() {
+      Cache.disableLocking();
+      tempDir =
+          fs.systemTempDirectory.createTempSync('sdk_validation_test').absolute;
+    });
+
+    tearDownAll(() {
+      tryToDelete(tempDir);
+      return server?.dispose();
+    });
+
+    testUsingContext('contains dart:ui', () async {
+      createSampleProject(tempDir, dartSource: '''
+import 'dart:ui' as ui;
+void main() {
+  // ignore: unnecessary_statements
+  ui.Window;
+}
+''');
+
+      await pubGet(context: PubContext.flutterTests, directory: tempDir.path);
+
+      server = AnalysisServer(dartSdkPath, <String>[tempDir.path]);
+
+      final int errorCount = await analyze(server);
+      expect(errorCount, 0);
+    });
+
+    testUsingContext('contains dart:html', () async {
+      createSampleProject(tempDir, dartSource: '''
+import 'dart:html' as html;
+void main() {
+  // ignore: unnecessary_statements
+  html.HttpStatus;
+}
+''');
+
+      await pubGet(context: PubContext.flutterTests, directory: tempDir.path);
+
+      server = AnalysisServer(dartSdkPath, <String>[tempDir.path]);
+
+      final int errorCount = await analyze(server);
+      expect(errorCount, 0);
+    });
+
+    testUsingContext('contains dart:js', () async {
+      createSampleProject(tempDir, dartSource: '''
+import 'dart:js' as js;
+void main() {
+  // ignore: unused_local_variable
+  var foo = js.allowInterop(null);
+}
+''');
+
+      await pubGet(context: PubContext.flutterTests, directory: tempDir.path);
+
+      server = AnalysisServer(dartSdkPath, <String>[tempDir.path]);
+
+      final int errorCount = await analyze(server);
+      expect(errorCount, 0);
+    });
+
+    testUsingContext('contains dart:js_util', () async {
+      createSampleProject(tempDir, dartSource: '''
+import 'dart:js_util' as js_util;
+void main() {
+  // ignore: unused_local_variable
+  var bar = js_util.jsify(null);
+}
+''');
+
+      await pubGet(context: PubContext.flutterTests, directory: tempDir.path);
+
+      server = AnalysisServer(dartSdkPath, <String>[tempDir.path]);
+
+      final int errorCount = await analyze(server);
+      expect(errorCount, 0);
+    });
+  });
+}
+
+void createSampleProject(Directory directory, {@required String dartSource}) {
+  final File pubspecFile =
+      fs.file(fs.path.join(directory.path, 'pubspec.yaml'));
+  pubspecFile.writeAsStringSync('''
+name: foo_project
+dependencies:
+  flutter:
+    sdk: flutter
+''');
+
+  final File dartFile =
+      fs.file(fs.path.join(directory.path, 'lib', 'main.dart'));
+  dartFile.parent.createSync();
+  dartFile.writeAsStringSync(dartSource);
+}
+
+Future<int> analyze(AnalysisServer server, {bool printErrors = true}) async {
+  int errorCount = 0;
+  final Future<bool> onDone =
+      server.onAnalyzing.where((bool analyzing) => analyzing == false).first;
```

Comment:
I think we were discussing on another issue that this was not exactly the right way to detect that the analysis server is finished reporting results. If the tool's `AnalysisServer` doesn't have a more correct way to detect the end of the error stream from the analysis server, could these tests become flaky?

Reply:
It's not part of the analysis server protocol spec, but it does work for all cases modulo when you're using analysis plugins. We do rely on this in other places in the source, including `lib/src/commands/analyze_continuously.dart` and analyze_once.dart.